apps/dashboard/src/components/engine/status/components/StatusCodes.tsx

```diff
@@ -0,0 +1,133 @@
+"use client";
+
+import { useQuery } from "@tanstack/react-query";
+import { Area, AreaChart, CartesianGrid, XAxis, YAxis } from "recharts";
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import {
+  type ChartConfig,
+  ChartContainer,
+  ChartLegend,
+  ChartLegendContent,
+  ChartTooltip,
+  ChartTooltipContent,
+} from "@/components/ui/chart";
+import type { JSX } from "react";
+
+export function StatusCodes() {
+  const monitorData = useQuery(["statusCodes"], async () => {
+    const res = await fetch("/api/engine-status/status-codes", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({}),
+    });
+
+    const json = await res.json();
+    return json;
+  });
+
+  const chartConfig: { [key: string]: { label: string; color: string } } = {
+    "200": {
+      label: "200",
+      color: "hsl(var(--chart-2))",
+    },
+    "207": {
+      label: "207",
+      color: "hsl(var(--chart-3))",
+    },
+    "404": {
+      label: "404",
+      color: "hsl(var(--chart-3))",
+    },
+    "500": {
+      label: "500",
+      color: "hsl(var(--chart-5))",
+    },
+    "502": {
+      label: "502",
+      color: "hsl(var(--chart-5))",
+    },
+    "524": {
+      label: "524",
+      color: "hsl(var(--chart-5))",
+    },
+    "N/A": {
+      label: "N/A",
+      color: "hsl(var(--chart-3))",
+    },
+  } satisfies ChartConfig;
+
+  const getAreaCharts = () => {
```

Comment:
again this can be ternary or in useMemo instead

Reply:
done